python-frontend/src/main/java/org/sonar/python/semantic/v2/ProjectLevelTypeTable.java

```diff
@@ -61,4 +62,30 @@ public ModuleType getModule(List<String> moduleNameParts) {
     }
     return parent;
   }
+
+  public PythonType getType(String typeFqn) {
+    return getType(typeFqn.split("\\."));
+  }
+
+  public PythonType getType(String... typeFqnParts) {
+    return getType(List.of(typeFqnParts));
+  }
+
+  public PythonType getType(List<String> typeFqnParts) {
+    var parent = (PythonType) rootModule;
+    for (int i = 0; i < typeFqnParts.size(); i++) {
```

Comment:
I feel we're repeating these loops quite a lot (I also have a few variations of them in my PR). I'm wondering if we could somehow generalize these FQN traversal somehow (no action required on this PR, just sharing my thoughts).